<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <let name="urdf_path" value="$(find-pkg-share mobile_robot_description)/urdf/mobile_robot.urdf.xacro" />

    <let name="bridge_config_path" value="$(find-pkg-share mobile_robot_bringup)/config/gz_bridge.yaml" />

    <let name="rviz_config_path" value="$(find-pkg-share mobile_robot_description)/rviz/default_config.rviz" />

    <let name="world_path" value="$(find-pkg-share mobile_robot_description)/worlds/Depot/model.sdf" />

    <node pkg="robot_state_publisher" exec="robot_state_publisher" >
        <param name="robot_description" value="$(command 'xacro $(var urdf_path)')"/>
    </node>

    <include file="$(find-pkg-share ros_gz_sim)/launch/gz_sim.launch.py" >
        <!-- <arg name="gz_args" value="empty.sdf -r" /> -->
         <arg name="gz_args" value="$(var world_path) -r" />
    </include>

    <node pkg="ros_gz_sim" exec="create" args="-topic robot_description" />

    <node pkg="ros_gz_bridge" exec="parameter_bridge">
        <param name="config_file" value="$(var bridge_config_path)" />
    </node>

    <node pkg="mobile_robot_bringup" exec="pointcloud_frame_id_converter" name="pointcloud_frame_id_converter_lidar" >
        <param name="frame_id" value="lidar_link" />
        <param name="subscription_topic" value="/lidar/pointcloud/points" />
        <param name="publisher_topic" value="lidar/pointcloud/points/corrected" />
    </node>

    <node pkg="mobile_robot_bringup" exec="pointcloud_frame_id_converter" name="pointcloud_frame_id_converter_depth_camera" >
        <param name="frame_id" value="depth_camera_link" />
        <param name="subscription_topic" value="/depth_camera/points" />
        <param name="publisher_topic" value="/depth_camera/points/corrected" />
    </node>

    <node pkg="rviz2" exec="rviz2" output="screen" args="-d $(var rviz_config_path)" />

</launch>